java/vector/src/test/java/org/apache/arrow/vector/ipc/BaseFileTest.java

```diff
@@ -846,4 +862,293 @@ protected void validateListAsMapData(VectorSchemaRoot root) {
+    try (ValueVector decoded = DictionaryEncoder.decode(encoded, dictionary)) {
+      Assertions.assertEquals(expected.length, encoded.getValueCount());
+      for (int i = 0; i < expected.length; i++) {
+        if (expected[i] == null) {
+          Assertions.assertNull(decoded.getObject(i));
+        } else {
+          assertNotNull(decoded.getObject(i));
+          Assertions.assertEquals(expected[i], decoded.getObject(i).toString());
+        }
+      }
+    }
+  }
+
+  protected void assertBlock(File file, int block, int state) throws Exception {
+    try (FileInputStream fileInputStream = new FileInputStream(file);
+         ArrowFileReader reader = new ArrowFileReader(fileInputStream.getChannel(), allocator);) {
+      reader.loadRecordBatch(reader.getRecordBlocks().get(block));
+      assertBlock(reader, block, state);
+    }
+  }
+
+  protected void assertBlock(ArrowReader reader, int block, int state) throws Exception {
+    VectorSchemaRoot r = reader.getVectorSchemaRoot();
+    FieldVector dictA = r.getVector("vectorA");
+    FieldVector dictB = r.getVector("vectorB");
+    FieldVector dictC = r.getVector("vectorC");
+    FieldVector dictD = r.getVector("vectorD");
+
+    switch (state) {
+      case 1:
+        assertDictionary(dictA, reader, valuesPerBlock.get(block)[0]);
+        assertNull(dictB);
+        assertNull(dictC);
+        assertNull(dictD);
+        break;
+      case 2:
+        assertNull(dictA);
+        assertDictionary(dictB, reader, valuesPerBlock.get(block)[1]);
+        assertNull(dictC);
+        assertNull(dictD);
+        break;
+      case 3:
+        assertDictionary(dictA, reader, valuesPerBlock.get(block)[0]);
+        assertDictionary(dictB, reader, valuesPerBlock.get(block)[1]);
+        assertNull(dictC);
+        assertNull(dictD);
+        break;
+      case 4:
+        assertNull(dictA);
+        assertNull(dictB);
+        assertDictionary(dictC, reader, valuesPerBlock.get(block)[2]);
+        assertNull(dictD);
+        break;
+      case 5:
+        assertDictionary(dictA, reader, valuesPerBlock.get(block)[0]);
+        assertNull(dictB);
+        assertDictionary(dictC, reader, valuesPerBlock.get(block)[2]);
+        assertNull(dictD);
+        break;
+      case 6:
+        assertDictionary(dictA, reader, valuesPerBlock.get(block)[0]);
+        assertDictionary(dictB, reader, valuesPerBlock.get(block)[1]);
+        assertDictionary(dictC, reader, valuesPerBlock.get(block)[2]);
+        assertNull(dictD);
+        break;
+      case 7:
+        assertNull(dictA);
+        assertNull(dictB);
+        assertNull(dictC);
+        assertDictionary(dictD, reader, valuesPerBlock.get(block)[3]);
+        break;
+      default:
+        throw new IllegalStateException("Unsupported state: " + state);
+    }
+  }
+
+  protected static Collection<Arguments> dictionaryParams() {
+    List<Arguments> params = new ArrayList<>();
+    for (int i = 1; i < 8; i++) {
```

Comment:
Not clear where these numbers come from.

Reply:
Just the number of unique test states. I'll drop a note.